patient_test.go

```diff
@@ -356,6 +358,188 @@ func TestPatientService_Update(t *testing.T) {
 	assert.NoError(err)
 }
 
+func TestPatientService_Update_empty_arrays(t *testing.T) {
+	testCases := map[string]struct {
+		consents   *NonNullJSONArray[*PatientConsent]
+		emails     *NonNullJSONArray[*PatientEmail]
+		insurances *NonNullJSONArray[*PatientInsuranceUpdate]
+		phones     *NonNullJSONArray[*PatientPhone]
+	}{
+		"pointers to nil slices": {
+			consents:   Ptr[NonNullJSONArray[*PatientConsent]](nil),
+			emails:     Ptr[NonNullJSONArray[*PatientEmail]](nil),
+			insurances: Ptr[NonNullJSONArray[*PatientInsuranceUpdate]](nil),
+			phones:     Ptr[NonNullJSONArray[*PatientPhone]](nil),
```

Comment:
Should these be pointers to a `nil` `NonNullJSONArray[*PatientConsent]` slice rather than a nil `Ptr` to a `NonNullJSONArray[*PatientConsent]` slice?

```suggestion
			consents:   Ptr(NonNullJSONArray[*PatientConsent](nil)),
			emails:     Ptr(NonNullJSONArray[*PatientEmail](nil)),
			insurances: Ptr(NonNullJSONArray[*PatientInsuranceUpdate](nil)),
			phones:     Ptr(NonNullJSONArray[*PatientPhone](nil)),
```

Reply:
The results appear to be the same according to spew. In both cases the value is non-nil but points to a nil slice. My understanding is that Go's generic handling is taking care of the casting of the nil since I'm specifying the type parameter.

Reply:
That being said, if you think your alternative is clearer, I'm happy to switch over to it.

Reply:
Ahh gotcha, yeah looking at this in the context of the implementation of `Ptr` this makes sense -- I do think it's a little bit clearer without the explicit type param, but not a big deal/worth changing